scripts/src/main/resources/scripts/command/python

```diff
@@ -0,0 +1,197 @@
+        sed -i -e 's/#import site/import site/g' "${PTH_FILE}"
+      else
+        doEcho "Error. Couldn't find _pth file."
+      fi
+      
+      # rename folder from Scripts to bin
+      if [ -d "${PYTHON_HOME}/Scripts" ]
+      then
+        mv "${PYTHON_HOME}"/Scripts "${PYTHON_HOME}"/bin
+      fi
+      
+    else
+      doInstallWithPackageManager "apt-get install -y make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev" \
+                                  "yum -y groupinstall \"Development Tools\"" \
+                                  "zypp install automake fdupes gcc gcc-c++ gcc-fortran gdbm-devel gettext-tools gmp-devel intltool libbz2-devel libexpat-devel libffi-devel libnsl-devel lzma-devel make ncurses-devel netcfg openssl-devel pkgconfig readline-devel sqlite-devel xz zlib-devel"
+      
+      
+      doInstall "-" "${PYTHON_HOME}" "python" "${PYTHON_VERSION}" "" "Python"
+      "${PYTHON_HOME}"/configure --enable-optimizations --with-ensurepip=install --prefix="${PYTHON_HOME}"
+      make -j 8
+      make altinstall
+      make clean
+      make distclean
+ 
+    fi
+    
+    if [ -d "${DEVON_IDE_HOME}/Python" ]
+    then
+      rm -r "${DEVON_IDE_HOME}/Python"
+    fi
+    
+  fi
+
+  # set bin locations
+  if ! doIsWindows
+  then
+    PYTHON_BIN=$(find "${PYTHON_SCRIPTS}" -name "python3.*" | sort | head -1)
+    PIP_BIN=$(find "${PYTHON_SCRIPTS}" -name "pip*" | sort | head -1)
+  else
+    PYTHON_BIN="${PYTHON_HOME}/python.exe"
+    PIP_BIN="${PYTHON_SCRIPTS}/pip.exe"
+    PIP3_BIN="${PYTHON_SCRIPTS}/pip3.exe"
+  fi
+ 
+  # check version
+  if [ "${1}" != "silent" ] && ! doIsQuiet
+  then
+    doRunCommand "${PYTHON_BIN} --version"
+  fi 
+  
+
+}
+
+function setPath() {
+  local path_bind="PATH=\$PATH:"
+  local set_path="${path_bind}${PYTHON_SCRIPTS}"
+  if ! grep -q "${set_path}" "${DEVON_IDE_HOME}/conf/devon.properties"
+    then
+      export "${set_path}"
+      echo -e "\n${set_path}" >> "${DEVON_IDE_HOME}/conf/devon.properties"
+      doEcho "Python and pip path is added to ${DEVON_IDE_HOME}/conf/devon.properties"
+  fi
+}
+
+function doRunPython() {
+  doSetup silent
+  doEcho "Running: ${PYTHON_BIN} ${*}"
+  if doIsQuiet
+  then
+    "${PYTHON_BIN}" "${@}" > /dev/null
+  else
+    "${PYTHON_BIN}" "${@}"
+  fi
+}
+
+function doRunPip() {
+  doSetup silent
+  if [ "${1}" = "pip" ]
+  then
+    doEcho "Running: ${*}"
+    shift
+    if doIsQuiet
+    then
+        doRunPython "${PIP_BIN}" "${@}" > /dev/null
+    else
+        doRunPython "${PIP_BIN}" "${@}"
+    fi
+  elif [ "${1}" = "pip3" ] && doIsWindows
+  then
+    doEcho "Running: pip3 ${*}"
+    if doIsQuiet
+    then
+      doRunPython "${PIP3_BIN}" "${@}" > /dev/null
+    else
+      doRunPython "${PIP3_BIN}" "${@}"
+   fi
+
+  fi
+}
+
+
+PYTHON_HOME="${DEVON_IDE_HOME}/software/python"
+PYTHON_SCRIPTS="${PYTHON_HOME}/bin"
```

Comment:
As this is now the `bin` folder in `Python` its name is kind of confusing.
Even more confusing for me is that there is also an additional variable `PYTHON_BIN` that does not point to the `bin` folder of Python but to the Python command executable.
I would suggest to name this variable `PYTHON_BIN` and either skip the `_BIN` suffix in the other variables (like done in other commandlets) or to use a different suffix like `_CMD` or `_EXE`.

Reply:
Excellent. @BizarreBits  thanks for addressing this. Now I do not get confused anymore. 👍 Done.